ckanext/security/mailer.py

```diff
@@ -58,8 +58,12 @@ def _build_footer_content(extra_vars):
         template = env.from_string(footer_content)
         return '\n\n' + template.render(**extra_vars)
     else:
-        footer_path = 'security/emails/lockout_footer.txt'
-        return '\n\n' + render_jinja2(footer_path, extra_vars)
+        if is_flask_request():
+            footer_path = 'security/emails/lockout_footer.txt'
```

Comment:
Why is this line inside the conditional?

And does the `render` function not work on all requests?

Reply:
You are correct in that the footer_path assignment should not be inside the conditional. Is the correct course of action for me to push another pull request with this fixed?

Regarding render vs render_jinja2, I believe that render is flask and render_jinja2 is pylons. This conditional pattern is repeated elsewhere in mailer.py - see for instance in the notify_lockout function that is also in mailer.py. 

That said, I am not 100% sure of the purpose of render_jinja2. I am using a 2.9.5 instance of CKAN and all I know is that render_jinja2 was causing the following KeyError exception in 2.9.5:

2022-09-02 14:30:54,586 INFO [ckanext.security.cache.login] bright locked out by brute force protection
2022-09-02 14:30:54,590 ERROR [ckanext.security.cache.login] Sending lockout notification for bright failed
Traceback (most recent call last):
File "/usr/lib/ckan/default/src/ckanext-security/ckanext/security/cache/login.py", line 75, in needs_lockout
notify_lockout(self.user, self.login_lock_timeout)
File "/usr/lib/ckan/default/src/ckanext-security/ckanext/security/mailer.py", line 86, in notify_lockout
+ _build_footer_content(extra_vars)
File "/usr/lib/ckan/default/src/ckanext-security/ckanext/security/mailer.py", line 62, in _build_footer_content
return '\n\n' + render_jinja2(footer_path, extra_vars)
File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 115, in render_jinja2
env = config['pylons.app_globals'].jinja_env
File "/usr/lib/ckan/default/src/ckan/ckan/common.py", line 98, in __getitem__
return self.store[key]
KeyError: 'pylons.app_globals'

Reply:
You can just push another commit to the branch.

I haven't looked into the render function differences in detail. If it works as-is, okay.